test/framework/src/main/java/org/elasticsearch/cluster/coordination/CoordinationStateTestCluster.java

```diff
@@ -148,14 +147,18 @@ void reboot() {
                 }
             }
 
-            localNode = DiscoveryNodeUtils.builder(localNode.getId())
-                .name(localNode.getName())
-                .ephemeralId(UUIDs.randomBase64UUID(random()))
-                .address(localNode.getHostName(), localNode.getHostAddress(), localNode.getAddress())
-                .attributes(localNode.getAttributes())
-                .roles(roles)
-                .version(localNode.getVersionInformation())
-                .build();
+            localNode = new DiscoveryNode(
```

Comment:
This is making a copy of a node - using the constructor explicitly makes this clearer, rather than the builder that can fill in random values we don't specify